### ISSUE TITLE
[Bug]: NaN Issue for gfx1151 - UNSLOTH_COMPILE_DISABLE

### ISSUE THREAD

**Issue by jrgosalvez** [amd/playbooks#611]
### Which playbook is this about?

Fine-tuning LLMs with Unsloth

### Operating system

Ubuntu

### Operating system version

24.04

### AMD hardware / platform

Ryzen xfx1151

### GPU driver / ROCm version

ROCm 7.2

### What happened?

Playbook was not optimized for gfx1151 silicon, resulting in NaN losses when weights loaded.

`UNSLOTH_COMPILE_DISABLE` does not resolve NaN issue on Ryzen gfx1151 for the AMD Playbook , while `attn_implementation="eager"` does work.

### What did you expect to happen?

Expected Playbook would run without kernel updates for AMD GPUs.

### Steps to reproduce

1. with a gfx1151 GPU 
2. update Ubuntu
3. ensure Unsloth is setup with ROCm wheels
4. download script
5. follow Playbook steps
6. see error

### Relevant logs, error messages, or screenshots

```shell

```

### Additional context

To Fix:

1. added user to groups for sudo priv.
2. updated ubuntu
3. uninstalled and reinstalled Unsloth for ROCm wheels
4. ran a series of diagnostics/tests/jobs to isolate weight/logit issue
5. cleared complied cache every run; even rebooted to ensure cache was clean
6. narrowed to Unsloth auto-compiler
7. UNSLOTH_COMPILE_DISABLE does not resolve NaN issue on Ryzen gfx1151 for the AMD Playbook , while attn_implementation="eager" does work

**Comment by harkgill-amd** [comment]
Hey @jrgosalvez, noticed your ROCm version is reported as 7.2 whereas the playbook installs ROCm 7.14 (recently updated from 7.13). Could you please give this a try with the PyTorch/ROCm wheels below,
```
python -m pip install --index-url https://repo.amd.com/rocm/whl-multi-arch/ "torch[device-gfx1151]==2.12.0+rocm7.14.0" "torchvision[device-gfx1151]==0.27.0+rocm7.14.0" "torchaudio==2.11.0+rocm7.14.0"
```
As an extra datapoint, gave this a run on my end with Strix Halo and the playbook ran correctly with no NaN losses. This is also passing in our CI - you can reference the exact steps it follows and results here https://github.com/amd/playbooks/actions/runs/29905966361/job/88877538447.

**Comment by harkgill-amd** [comment]
CI tests are passing with valid loss values for gfx1151 + `Fine-tuning LLMs with Unsloth` https://github.com/amd/playbooks/actions/runs/30252884496/job/89934850981.

If upgrading to ROCm 7.14 does not resolve the NaN loss on your end, please leave a comment and I'll reopen this issue.
